src/app/app.component.ts

```diff
@@ -12,12 +12,15 @@ export class AppComponent implements OnInit {
 
   constructor(private readonly mainService: MainService) {}
 
-  ngOnInit() {
-    (async () => {
-      this.elements = await this.mainService.getAllElements().toPromise();
-    })();
+  ngOnInit(): void {
+     this.mainService.getAllElements().toPromise().then(
```

Comment:
instead of IIFE function, converted to normal sync code and handled value inside `.then` and error can also be captured inside `.catch` but still i'd prefer to use `observable` here so the component can have chance to cancel `async` task when it get disposed ensuring mem leak. But, since this is the root file, i think it doesn't required to refactor entirely as was concern about capturing the exception using either of `.catch` from promise or `error` prop from `observable` to avoid the use of `try,catch` block inside `IIFE` :smile: